### PR TITLE
feat(rpc): allowing all CORS origins

### PR DESCRIPTION
**Pull request by distractedm1nd**
Needed for API Playground

### CODE REVIEW

**Review by cristaloleg — APPROVED** [review]
SGTM but I haven't looked deeply enough.

**Comment by walldiss** [comment]
Coverted to draft, to prevent merge before it will get updated

File: api/rpc/server.go [hunk 1]
```diff
@@ -44,7 +44,19 @@ func NewServer(address, port string, authDisabled bool, secret jwt.Signer) *Serv
 	}
 	srv.srv.Handler = &auth.Handler{
 		Verify: srv.verifyAuth,
-		Next:   rpc.ServeHTTP,
+		Next: func(w http.ResponseWriter, r *http.Request) {
+			if origin := r.Header.Get("Origin"); origin != "" {
+				w.Header().Set("Access-Control-Allow-Origin", origin)
```
Comment (Wondertan) [inline api/rpc/server.go:49]: Any reason you dont set simply a `*`? AFAIU, it eill have the same effect

**Review by Wondertan — COMMENTED** [review]
Ideally we would add a config section for the rpc server http header where you would configure the playground node to allow all origins, instead of every node to allow that

**Comment by walldiss** [comment]
After discussing with @distractedm1nd I've put into the issue the approach that will be implemented. Current one is irrelevant
https://github.com/celestiaorg/celestia-node/issues/3481

File: api/rpc/server.go (hunk 1 above)
Comment (distractedm1nd) [reply api/rpc/server.go:49]: Doesn't work. Was my first thought too, but it had something to do with the OPTIONS request IIRC. I'll post a link tomorrow when i'm back at the computer

**Pull request closed without merging**